plugins/chat/assets/javascripts/discourse/models/chat-message.js

```diff
@@ -1,26 +1,193 @@
-import RestModel from "discourse/models/rest";
 import User from "discourse/models/user";
-import EmberObject from "@ember/object";
+import { cached, tracked } from "@glimmer/tracking";
+import { TrackedArray, TrackedObject } from "@ember-compat/tracked-built-ins";
+import ChatMessageReaction from "discourse/plugins/chat/discourse/models/chat-message-reaction";
+import Bookmark from "discourse/models/bookmark";
+import I18n from "I18n";
+import guid from "pretty-text/guid";
 
-export default class ChatMessage extends RestModel {}
+export default class ChatMessage {
+  static cookFunction = null;
 
-ChatMessage.reopenClass({
-  create(args = {}) {
-    this._initReactions(args);
-    this._initUserModel(args);
+  static create(channel, args = {}) {
+    return new ChatMessage(channel, args);
+  }
 
-    return this._super(args);
-  },
+  static createStagedMessage(channel, args = {}) {
+    args.staged_id = guid();
+    return new ChatMessage(channel, args);
+  }
 
-  _initReactions(args) {
-    args.reactions = EmberObject.create(args.reactions || {});
-  },
+  @tracked id;
```

Comment:
They really need a `@trackAllOfThisStuff(a, b, c)` API :laughing:

Reply:
could be handy indeed